src/common/cockpitmemory.h

```diff
@@ -28,39 +28,6 @@ G_BEGIN_DECLS
 void     cockpit_secclear                (gpointer data,
                                           gssize length);
 
-#define DEFINE_CLEANUP_FUNCTION(Type, name, func) \
-  static inline void name (void *v) \
-  { \
-    func (*(Type*)v); \
-  }
-
-#define DEFINE_CLEANUP_FUNCTION0(Type, name, func) \
-  static inline void name (void *v) \
-  { \
-    if (*(Type*)v) \
-      func (*(Type*)v); \
-  }
-
-/* These functions shouldn't be invoked directly;
- * they are stubs that:
- * 1) Take a pointer to the location (typically itself a pointer).
- * 2) Provide %NULL-safety where it doesn't exist already (e.g. g_object_unref)
- */
-DEFINE_CLEANUP_FUNCTION0(GHashTable*, local_hashtable_unref, g_hash_table_unref)
-DEFINE_CLEANUP_FUNCTION0(GObject*, local_obj_unref, g_object_unref)
-DEFINE_CLEANUP_FUNCTION0(GVariant*, local_variant_unref, g_variant_unref)
-DEFINE_CLEANUP_FUNCTION0(GVariantIter*, local_variant_iter_free, g_variant_iter_free)
-
-DEFINE_CLEANUP_FUNCTION(char**, local_strfreev, g_strfreev)
-DEFINE_CLEANUP_FUNCTION(void*, local_free, g_free)
-
-#define cleanup_free __attribute__ ((cleanup(local_free)))
-#define cleanup_unref_object __attribute__ ((cleanup(local_obj_unref)))
-#define cleanup_unref_variant __attribute__ ((cleanup(local_variant_unref)))
-#define cleanup_free_variant_iter __attribute__ ((cleanup(local_variant_iter_free)))
-#define cleanup_unref_hashtable __attribute__ ((cleanup(local_hashtable_unref)))
-#define cleanup_strfreev __attribute__ ((cleanup(local_strfreev)))
```

Comment:
Agreed, glib 2.44 provides the `g_auto` stuff which is much more elegant, so once we can bump the dependency we are going to use those instead of these custom definitions.